### DISCUSSION TITLE
Replacing PowerShell scripts by vanilla CMake

### DISCUSSION THREAD

**Discussion by j-jorge** [axmolengine/axmol#2090] [General]
Hi, Axmol relies on PowerShell scripts to download some dependencies. The process is smooth, but only if PowerShell is available!

I occasionally work on a platform where this tool is not available (arm64 Android). There is no way to run the prebuilt binaries there, and no way to build pwsh from source since it requires pwsh to bootstrap the process!

A solution based on Python seems more portable (I have Python installed on my device), but the best would be a pure CMake solution.

What do you think? How difficult would it be to replace these scripts? Is it worth the effort?

**Comment by smilediver** [comment]
I also don't like the use of PowerShell and would prefer CMake/Python scripts. Our IT guy got a heart attack when I told him, that we'll need PowerShell on macOS build machines. :D

And I think Axmol is overdoing the whole setup stage. I don't really want it to download and install all the requirements like CMake itself, PowerShell, VS compiler, Android SDK, etc. I wish this whole setup stage was split into separate steps and they were more independent and optional, giving more control to the user. The part where it modifies environment variables is particularly annoying.

**Reply by GamesStudioTwo** [reply]
Yeah it was a shock to me to but the old script would only run on python 2 and not 3 version so it kind of dose away with my problem I need to use the better python 3 as I use a script to cut out tile's from a .png image and wont work in version 2 so I'm happy with using the PowerShell. as it work's out perfect that I can have Python 3 installed else I would have to downgrade every time I make a new project.

**Reply by smilediver** [reply]
No one is arguing to use old Python 2. That would also be a bad idea, since it's not supported or maintained anymore.
